node/src/rpc/evm.rs

```diff
@@ -39,8 +39,24 @@ use sp_core::H256;
 use sp_inherents::CreateInherentDataProviders;
 use sp_runtime::traits::{BlakeTwo256, Block as BlockT};
 
+pub struct CentrifugeEthConfig<B, C, BE>(std::marker::PhantomData<(B, C, BE)>);
+impl<B, C, BE> fc_rpc::EthConfig<B, C> for CentrifugeEthConfig<B, C, BE>
+where
+	B: BlockT,
+	C: sc_client_api::StorageProvider<B, BE> + Sync + Send + 'static,
+	BE: Backend<B> + 'static,
+{
+	// This type is intended to override (i.e. adapt) evm calls to precompiles for proper gas estimation.
+	//
+	// NOTE: Not used by our precompiles right now. Therefore, no need to provide impl.
+	type EstimateGasAdapter = ();
+	// Assumes the use of HashedMapping<BlakeTwo256> for address mapping
+	type RuntimeStorageOverride =
```

Comment:
What is this type doing?

Reply:
AFAIU, the eth config serves as an override adapter for `eth_call` similar to [geth](https://geth.ethereum.org/docs/interacting-with-geth/rpc/ns-eth). 

I have to admit that I am not confident in using the `SystemAccountId32StorageOverride` ([-> impl](https://github.com/moonbeam-foundation/frontier/blob/7f424fb9fc1ca9158a59be54cb183beaff9f799a/client/rpc/src/lib.rs#L140-L141)) type or whether `()` (i.e. no override) is the way to go. This was inspired [by Astar](https://github.com/AstarNetwork/Astar/blob/111d18fbfba6076570cb6454573f88afe0ee53b3/bin/collator/src/rpc.rs#L103).

WDYT @mustermeiszer?

Using the `ethereumRuntimeRPCApi` runtime API, I could not notice any difference between using `SystemAccountId32StorageOverride` and `()`. However, I did not execute any EVM call.